lib/modules/profile/screens/auth_gate.dart

```diff
@@ -1,11 +1,17 @@
 import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
 
 import 'profile.dart';
 import 'sign_in.dart';
 
 class AuthGate extends StatelessWidget {
-  const AuthGate({Key? key}) : super(key: key);
+  const AuthGate({
+    this.openProfile = true,
+    Key? key,
+  }) : super(key: key);
+
+  final bool openProfile;
```

Comment:
I don't understand this field. Why would you not want to open the profile?

Reply:
When user hasn't signed in and taps "like", they're suggested to auth. After such login I return user to the old screen instead of sending them into the profile page.

Reply:
Then I'd rename it to something like `popAfterLogin` (default `false`) since that's the exception to the rule